File: core/src/main/java/com/animation/morphtransitions/IntProperty.java
package com.animation.morphtransitions;

import android.util.Property;

abstract class IntProperty<T> extends Property<T, Integer> {

    public IntProperty(String name) {
        super(Integer.class, name);
    }

    /**
     * A type-specific override of the {@link #set(Object, Integer)} that is faster when dealing
     * with fields of type <code>int</code>.
     */
    public abstract void setValue(T object, int value);

    @Override
    final public void set(T object, Integer value) {
        setValue(object, value);
    }

}

File: core/src/main/java/com/animation/morphtransitions/IntProperty.kt
package com.animation.morphtransitions

import android.util.Property

internal abstract class IntProperty<T>(name: String)
    : Property<T, Int>(Int::class.java, name) {

    /**
     * A type-specific override of the [.set] that is faster when dealing
     * with fields of type `int`.
     */
    abstract fun setValue(obj: T, value: Int)

    override fun set(obj: T, value: Int) {
        setValue(obj, value)
    }
}
